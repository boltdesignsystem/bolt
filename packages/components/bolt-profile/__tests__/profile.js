// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
// import schema from '../profile.schema';
// const { [PROP KEY], [PROP KEY]... } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    name: {
      content: '<strong>First Last</strong> (@username)',
      tag: 'h2',
    },
    job_title: 'Job Title',
    location: 'Location',
    stats: [
      {
        number: '15',
        label: 'Achievements',
        attributes: {
          type: 'button',
        },
      },
      {
        number: '3k',
        label: 'Contributions',
        attributes: {
          href: '#!',
        },
      },
    ],
    message: {
      label: 'Send message',
      attributes: {
        href: '#!',
      },
    },
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Profile component', () => {
  test('default', async () => {
    const results = await render('@bolt-components-profile/profile.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

// describe('Bolt Profile prop -', () => {
//   // Target each of the schema keys with the following pattern
//   [PROP KEY].enum.forEach(async (option) => {
//     test(`[PROP KEY] items: ${option}`, async () => {
//       const results = await render(
//         '@bolt-components-profile/profile.twig',
//         {
//           ...fixtures.defaultData,
//           [PROP KEY]: option,
//         }
//       );

//       await expect(results.ok).toBe(true);
//       await expect(results.html).toMatchSnapshot();
//     });
//   });
// });
