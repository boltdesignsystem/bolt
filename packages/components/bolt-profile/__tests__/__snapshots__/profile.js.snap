// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Profile component default 1`] = `
<div class="c-bolt-profile">
  <div class="c-bolt-profile__cover">
    <img src="/images/backgrounds/navy-light-circles.jpg"
         sizes="100vw"
         width="2560"
         height="2560"
         loading="eager"
         class="e-bolt-image e-bolt-image--bg"
    >
  </div>
  <div class="c-bolt-profile__avatar">
    <svg viewbox="0 0 500 500"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path d="m450 398.32c-41.11-48-92.49-77.33-155.45-88.88 35.52-18.24 57.86-45.7 62.39-85s-9.7-72-40.76-96.63c-46.91-37.19-117.18-25.81-151.61 23.46-34.75 49.67-22.45 125.53 44.2 158.24-30.77 5.63-59.25 15.3-85.67 30.49a255.37 255.37 0 0 0 -69.48 58.26c-61.89-78.26-75.62-216.47 17.11-316.44 89.95-96.95 242.27-106.66 344.33-16.89 98.08 86.23 110.86 231.6 34.94 333.39z"
            fill="var(--m-bolt-text-on-disabled)"
      >
      </path>
      <g fill="var(--m-bolt-disabled)">
        <path d="m442.17 407.77c-94.44 118.39-282.52 121.89-380.88.38 88.1-113.4 277.05-126.38 380.88-.38z">
        </path>
        <path d="m345.69 211.49c-.36 52.58-42.88 95-94.65 94.52-51.55-.52-94-43.31-93.51-94.17.57-53.33 43-95.3 96.46-94.12 52.65 1.17 91.87 44.79 91.7 93.77z">
        </path>
      </g>
    </svg>
  </div>
  <div class="c-bolt-profile__user-info">
    <div class="c-bolt-profile__user-headline">
      <h1>
        <strong>
          First Last
        </strong>
        (@username)
      </h1>
    </div>
    <div class="c-bolt-profile__user-meta">
      Job Title
      <br>
      <svg class="e-bolt-icon"
           xmlns="http://www.w3.org/2000/svg"
           viewbox="0 0 32 32"
           aria-hidden="true"
      >
        <path fill="#151619"
              d="M16 0C8.6 0 2.7 6 2.7 13.3c.1 2.6.8 5.1 2 7.3 1.2 2.1 2.6 4.1 4.3 5.8 1.9 2 4 3.8 6.2 5.3.5.3 1 .3 1.5 0 2.2-1.6 4.3-3.4 6.2-5.3 1.7-1.7 3.1-3.7 4.3-5.8 1.3-2.2 2-4.7 2-7.3C29.3 6 23.4 0 16 0zm0 18.7c-2.9 0-5.3-2.4-5.3-5.3S13.1 8 16 8s5.3 2.4 5.3 5.3-2.4 5.4-5.3 5.4z"
        >
        </path>
      </svg>
      Location
    </div>
  </div>
</div>
`;
